driver/configurations/bazel.cmake

```
@@ -80,7 +80,13 @@ else()
 endif()
 
 if(DASHBOARD_PROCESSOR_COUNT GREATER 1)
-  math(EXPR DASHBOARD_JOBS "7 * ${DASHBOARD_PROCESSOR_COUNT} / 8")
+  # NOTE: linux clang address sanitizer builds experience spurious errors, see
+  # https://github.com/RobotLocomotion/drake/issues/17560
+  if(NOT APPLE AND COMPILER STREQUAL "clang" AND DASHBOARD_JOB_NAME MATCHES "address-sanitizer")
+    math(EXPR DASHBOARD_JOBS "7 * ${DASHBOARD_PROCESSOR_COUNT} / 16")
```

Pedantic: the old logic ensured that this would never be 0; should we retain that?
```suggestion
    if(DASHBOARD_PROCESSOR_COUNT GREATER 2)
      math(EXPR DASHBOARD_JOBS "7 * ${DASHBOARD_PROCESSOR_COUNT} / 16")
    else()
      set(DASHBOARD_JOBS 1)
    endif()
```